src/OrleansCodeGenerator/SerializerGenerator.cs

```diff
@@ -595,12 +595,12 @@ public ExpressionSyntax GetGetter(ExpressionSyntax instance, ExpressionSyntax se
                     && this.FieldInfo.FieldType.GetTypeInfo().IsInterface
                     && !typeof(IGrainObserver).IsAssignableFrom(this.FieldInfo.FieldType))
                 {
-                    var getAsReference = getValueExpression.Member(
+                    var getAsReference = getValueExpression.ConditionalMember(
                         (IAddressable grain) => grain.AsReference<IGrain>(),
                         this.FieldInfo.FieldType);
 
                     // If the value is not a GrainReference, convert it to a strongly-typed GrainReference.
-                    // C#: !(value is GrainReference) ? value.AsReference<TInterface>() : value;
+                    // C#: !(value is GrainReference) ? value?.AsReference<TInterface>() : value;
```

Comment:
*I believe* we support users building their Orleans apps without c#6... Orleans development with pre-6 is not supported.